cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -194,8 +194,13 @@ where
 			self.client.runtime_api().intermediate_roots(&BlockId::Hash(parent_hash))?;
 		roots.push(state_root.encode());
 
-		let trace_root =
-			BlakeTwo256::ordered_trie_root(roots.clone(), sp_core::storage::StateVersion::V1);
+		let trace_root = crate::merkle_tree::MerkleTree::new(
+			roots
+				.iter()
+				.map(|r| r.as_slice().try_into().expect("Storage root type is [u8; 32]; qed")),
```

Comment:
This doesn't give me a lot of confidence. The whole flow throughout is just `Vec<u8>` and only at the last step transformed into `[u8; 32]`. Can we make intermediate roots `[u8; 32]` in the first place instead? That way if there is a mismatch, it will be uncovered much sooner.